app/api/history/views.py

```diff
@@ -0,0 +1,38 @@
+from api.history.schemas import GetHistory
+from api.history.services import HistoryService
+from db.db import db_session
+from db.models.history import History
+from fastapi import APIRouter, Depends, Query
+from sqlmodel.ext.asyncio.session import AsyncSession
+import fastapi_users
+
+
+current_user = fastapi_users.current_user()
+
+router = APIRouter(
+    dependencies= [Depends(current_user)]
+)
+
+
+@router.get("/user/history", response_model=list[GetHistory])
+async def get_user_history(offset: int = 0, limit: int = Query(default=20, lte=20), session: AsyncSession = Depends(db_session)) -> list[History]:
+    user_history = HistoryService(session=session)
+    user = current_user
+    return await user_history.get_user_history(user_id= user.id, limit=limit, offset=offset)
+
+
+@router.delete("/remove/history", response_model=GetHistory)
+async def remove_all_history(session: AsyncSession = Depends(db_session)):
```

Comment:
I can't seem to think of a situation where we would want to allow deletion of the history records from the outside. Could you give some insight into your decision to support deletion?